### PR TITLE
SANS Reduced unit test file dependencies

### LINKED ISSUE #22099: Remove SANS unit test file dependencies.
Several of the SANS unit tests are intermittently failing due to file access issues. This should be cleaned up if possible. 

The error messages produced by the SANS backend are also often unhelpful so these should be amended if possible.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
This pull requests slightly re factors the sans state object so its creation is less dependent on IOS operations. Relevant unit tests are then changed so that they rely on a mock file information object rather than creating a file information object themselves. 

**To test:*
Review the code and confirm that all the unit tests pass.

Fixes #22099. 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
@Mantid-Matthew It looks like there are some system test failures.

File: scripts/SANS/sans/state/data.py [hunk 1]
```diff
@@ -106,10 +105,10 @@ def validate(self):
 # ----------------------------------------------------------------------------------------------------------------------
 # Builder
 # ----------------------------------------------------------------------------------------------------------------------
-def set_information_from_file(data_info):
-    file_name = data_info.sample_scatter
-    file_information_factory = SANSFileInformationFactory()
-    file_information = file_information_factory.create_sans_file_information(file_name)
+def set_information_from_file(data_info, file_information):
+    # file_name = data_info.sample_scatter
```
Comment (martyngigg) [inline scripts/SANS/sans/state/data.py]: Can we remove the commented-out code?

File: scripts/SANS/sans/state/data.py [hunk 2]
```diff
@@ -134,17 +134,17 @@ def build(self):
         # There are some elements which need to be read from the file. This is currently:
```
Comment (martyngigg) [inline scripts/SANS/sans/state/data.py]: Does this comment imply that reading will still be done here? Am I to understand correctly that the file reading has actually now already been done?

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: scripts/SANS/sans/state/data.py (hunk 2 above)
Comment (Mantid-Matthew) [reply scripts/SANS/sans/state/data.py]: Yes the reading has now already been done outside the state builder function. I've updated the comment to make this clearer.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**